### PR TITLE
feat(unify): show spec runner header in run mode

### PR DESCRIPTION
**Pull request by marktnoonan**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-951

### Additional details

This PR follows the existing pattern of splitting some components up for run mode, so that we now have SpecRunnerHeaderOpenMode and SpecRunnerHeaderRunMode components. The RunMode version avoids any dependency on GraphQL.

It is using our existing HideDuringScreenshot wrapper component. I think there may be some extra complexity around this, and testing needed, but that will come in a full review of the screenshot API behavior and edge cases: https://cypress-io.atlassian.net/browse/UNIFY-1078

Most of the code changes are new tests for the appearance and (lack of) behavior of the AUT header in run mode conditions. In e2e tests, run mode is simulated on the client side - I'd really like to do this in the same way we open projects in open mode, but it seemed outside the scope of this ticket. Still, if there is a better option than my `simulateRunModeInUI` function, let's switch to that.

### How has the user experience changed?

Run mode now looks like this:

Header shown in run mode

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/frontend-shared/src/graphql/urqlClient.ts
Comment (elevatebart) [inline packages/frontend-shared/src/graphql/urqlClient.ts:44]: Didn't we already have a Cypress.platform and Cypress.browser. Why do we need this extra variable?

**Review by elevatebart — DISMISSED** [review]
One small remark

File: packages/frontend-shared/src/graphql/urqlClient.ts
Comment (marktnoonan) [reply packages/frontend-shared/src/graphql/urqlClient.ts:44]: Good point, I wasn't aware of Cypress.browser, I'll take a look and if that works, I'll get rid of this global

File: packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts
Comment (tbiethman) [inline packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts:25]: I know the styles in these snapshots aren't complete, but even as-is you'll run into non-determinant UI with the reporter duration:
Screen Shot 2022-02-15 at 1 48 42 PM

Though I don't even see the unstyled header markup in that snapshot: https://percy.io/cypress-io/cypress/builds/15753852/changed/889019696?browser=chrome&browser_ids=20&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1200&widths=250%2C300%2C400%2C500%2C600%2C800%2C900%2C1000%2C1032%2C1200%2C1280

Comment (tbiethman) [reply packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts:25]: We should consider removing these snapshots until [UNIFY-1150](https://cypress-io.atlassian.net/browse/UNIFY-1150) can be completed.

Comment (marktnoonan) [reply packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts:25]: I wonder if it's better to leave them in. They can be approved for now, and when the CSS issue is fixed, these would show up again in the diff automatically and get reviewed. If we make some other way to automatically force these back in (maybe skip for a week based on Date.now()?) I'm ok with that too.

Comment (lmiller1990) [reply packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts:25]: We have a pattern for this in `reporter` already, we customize the Percy command to remove non-deterministic elements: https://github.com/cypress-io/cypress/blob/73e936e28992d5f533a8341a923dd61719bd12d5/packages/reporter/cypress/support/e2e.ts#L9-L12

File: packages/app/cypress/e2e/cypress-in-cypress-e2e.cy.ts
Comment (lmiller1990) [inline packages/app/cypress/e2e/cypress-in-cypress-e2e.cy.ts:86]: I wonder if we want to make this a custom command, or at least put in it a function explaining what it does (I have no idea why this is necessary).

**Review by lmiller1990 — COMMENTED** [review]
I have one comment!

File: packages/app/cypress/e2e/cypress-in-cypress-e2e.cy.ts
Comment (marktnoonan) [reply packages/app/cypress/e2e/cypress-in-cypress-e2e.cy.ts:86]: Oh yes - I was just trying some stuff out tonight after talking with Tyler. Let me convert this to draft to avoid confusion. Percy snapshots from cypress-in-cypress tests don't have any styles from the Vue app. I'm hoping that something like this, to get it working, will reveal a bit more about what a permanent solution would look like. Probably it can be solved in a way that doesn't need this kind of thing.

Comment (marktnoonan) [reply packages/app/cypress/e2e/cypress-in-cypress-e2e.cy.ts]: Removed now!

File: packages/frontend-shared/src/graphql/urqlClient.ts
Comment (marktnoonan) [reply packages/frontend-shared/src/graphql/urqlClient.ts:44]: I did implement this, `Cypress` is not available right away when the setup function runs, we have to wait for it, which leads to polling for it. I'm going to leave it as is, I think it's a little more simple & reliable this way, plus header can render without even the tiny few ms extra delay.

File: packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts
Comment (tbiethman) [reply packages/app/cypress/e2e/cypress-in-cypress-run-mode.cy.ts:25]: It's also possible to send elementOverrides on a per-snapshot basis, but you have to work around the typing:
```ts
  // @ts-ignore
  cy.percySnapshot({
    elementOverrides: {
      '.runnable-header .duration': ($el) => {
        $el.text('XX:XX')
      },
    },
  })
```

But these and the existing snapshots are going to need a lot of work once UNIFY-1150 completes. So long as we're not tripping over them in Percy until that work lands, I don't have a strong opinion whether they stay or go. If they do go, we should log an issue to make sure they're not lost.

**Review by tbiethman — APPROVED** [review]
Gave it a test drive locally, looks great :+1:

File: packages/app/src/runner/SpecRunnerHeaderRunMode.vue
Comment (lmiller1990) [inline packages/app/src/runner/SpecRunnerHeaderRunMode.vue:5]: Seems `props.width` is optional? If so, this needs a conditional, or we might render `width: px`. My suspicion is `width` is  **not** optional.

**Review by lmiller1990 — COMMENTED** [review]
One comment!

File: packages/frontend-shared/src/graphql/urqlClient.ts
Comment (elevatebart) [reply packages/frontend-shared/src/graphql/urqlClient.ts:44]: OK, good to know

File: packages/app/src/runner/SpecRunnerHeaderRunMode.vue
Comment (marktnoonan) [reply packages/app/src/runner/SpecRunnerHeaderRunMode.vue:5]: true made required 👍

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**
